src/kv_manager/error.rs

```diff
@@ -40,7 +40,9 @@ pub enum InnerKvError {
     SledErr(#[from] encrypted_sled::Error),
     #[error("Logical Error: {0}")] // Logical errors (eg double deletion)
     LogicalErr(String),
-    #[error("Bincode Error: {0}")] // (De)serialization errors
-    BincodeErr(#[from] bincode::Error),
+    #[error("Serialization Error: {0}")] // Serialization errors
```

Comment:
If error messages are static, we can lose the "(`String`)" and put the message directly in here.